src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,206 @@
+import {
+  FC, useEffect, useState,
+} from 'react';
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from '../../api/api';
+import { Todo } from '../../type';
+import { Footer } from '../Footer';
+import { TodoList } from '../TodoList';
+
+const useLocalStorage = () => {
+  const todosFromLocal = localStorage.getItem('todos');
+
+  try {
+    return todosFromLocal ? JSON.parse(todosFromLocal) : [];
+  } catch (error) {
+    return [];
+  }
+};
+
+export const TodoApp: FC = () => {
+  const [targetValue, setTargetValue] = useState('');
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [useServer, setUseServer] = useState(true);
+
+  async function load() {
+    const todoFromServer = await getTodos();
+
+    setVisibleTodos(todoFromServer);
+  }
+
+  useEffect(() => {
+    if (useServer) {
+      load();
+    } else {
+      setVisibleTodos(useLocalStorage);
+    }
+  }, [useServer]);
+
+  useEffect(() => {
+    if (useServer) {
+      load();
+    } else {
+      setVisibleTodos(useLocalStorage);
+    }
+  }, []);
+
+  const addNewTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+    const newTodo: Todo = {
+      title: targetValue,
+      completed: false,
+      userId: 888,
+    };
+
+    if (useServer) {
+      addTodo(newTodo);
+    }
+
+    setVisibleTodos(prevTodo => [...prevTodo,
+      { id: +new Date(), ...newTodo },
+    ]);
+
+    setTargetValue('');
+  };
+
+  const onDelete = (id: number | undefined) => {
+    setVisibleTodos(state => {
+      return state.filter(todoForChange => {
+        if (useServer && id) {
+          deleteTodo(id);
+        }
+
+        return todoForChange.id !== id;
+      });
+    });
+  };
+
+  const DeleteAll = () => {
+    setVisibleTodos(prevTodo => {
+      return prevTodo.filter(todo => {
+        if (useServer && todo.completed && todo.id) {
+          deleteTodo(todo.id);
+        }
+
+        return !todo.completed;
+      });
+    });
+  };
+
+  const CompletedAll = () => {
```

Comment:
USE camelCase for naming functions
```suggestion
  const completeAllTodos = () => {
```
```suggestion
  const CompletedAll = () => {
```